modular_citadel/code/game/machinery/toylathe.dm

```diff
@@ -0,0 +1,389 @@
+#define AUTOYLATHE_CATEGORY_MENU   2
+#define AUTOYLATHE_SEARCH_MENU     3
+
+/obj/machinery/autoylathe
+	name = "autoylathe"
+	desc = "It produces items using plastic, metal and glass."
+	icon_state = "autolathe"
+	density = TRUE
+	anchored = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 10
+	active_power_usage = 100
+	circuit = /obj/item/circuitboard/machine/autoylathe
+
+	var/operating = FALSE
+	var/list/L = list()
+	var/list/LL = list()
+	var/hacked = FALSE
+	var/disabled = 0
+	var/shocked = FALSE
+	var/hack_wire
+	var/disable_wire
+	var/shock_wire
+
+	var/busy = FALSE
+	var/prod_coeff = 1
+
+	var/datum/design/being_built
+	var/datum/techweb/stored_research
+	var/list/datum/design/matching_designs
+	var/selected_category
+	var/screen = 1
+
+	var/list/categories = list(
+							"Toys",
+							"Figurines",
+							"Pistols",
+							"Rifles",
+							"Heavy",
+							"Melee",
+							"Armor",
+							"Adult",
+							"Misc",
+							"Imported"
+							)
+
+/obj/machinery/autoylathe/Initialize()
+	AddComponent(/datum/component/material_container, list(MAT_METAL, MAT_GLASS, MAT_PLASTIC), 0, TRUE, null, null, CALLBACK(src, .proc/AfterMaterialInsert))
+	. = ..()
+
+	wires = new /datum/wires/autoylathe(src)
+	stored_research = new /datum/techweb/specialized/autounlocking/autoylathe
+	matching_designs = list()
+
+/obj/machinery/autoylathe/Destroy()
+	QDEL_NULL(wires)
+	return ..()
+
+/obj/machinery/autoylathe/interact(mob/user)
+	if(!is_operational())
+		return
+
+	if(shocked && !(stat & NOPOWER))
+		shock(user,50)
+
+	var/dat
+
+	switch(screen)
+		if(AUTOYLATHE_MAIN_MENU)
+			dat = main_win(user)
+		if(AUTOYLATHE_CATEGORY_MENU)
+			dat = category_win(user,selected_category)
+		if(AUTOYLATHE_SEARCH_MENU)
+			dat = search_win(user)
+
+	var/datum/browser/popup = new(user, "Autoylathe", name, 400, 500)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/autoylathe/on_deconstruction()
+	GET_COMPONENT(materials, /datum/component/material_container)
+	materials.retrieve_all()
+
+/obj/machinery/autoylathe/attackby(obj/item/O, mob/user, params)
+	if (busy)
+		to_chat(user, "<span class=\"alert\">The autoylathe is busy. Please wait for completion of previous operation.</span>")
+		return TRUE
+
+	if(default_deconstruction_screwdriver(user, "autolathe_t", "autolathe", O))
+		updateUsrDialog()
+		return TRUE
+
+	if(exchange_parts(user, O))
+		return TRUE
+
+	if(default_deconstruction_crowbar(O))
+		return TRUE
+
+	if(panel_open && is_wire_tool(O))
+		wires.interact(user)
+		return TRUE
+
+	if(user.a_intent == INTENT_HARM) //so we can hit the machine
+		return ..()
+
+	if(stat)
+		return TRUE
+
+	if(istype(O, /obj/item/disk/design_disk))
+		user.visible_message("[user] begins to load \the [O] in \the [src]...",
+			"You begin to load a design from \the [O]...",
+			"You hear the chatter of a floppy drive.")
+		busy = TRUE
+		var/obj/item/disk/design_disk/D = O
+		if(do_after(user, 14.4, target = src))
+			for(var/B in D.blueprints)
```

Comment:
you sure you want these things to be able to print all uploaded designs? doesn't make sense for a toy lathe to print pico manips.
if you do though i don't really care, this isn't mandatory and i'm just nitpicking at this point.

Reply:
Intentional. It can manipulate glass and metal as well, no reason it shouldn't be able to make manips if someone takes the time to upload a disk.